common/src/main/java/com/wynntils/mc/mixin/HumanoidArmorLayerMixin.java

```diff
@@ -0,0 +1,27 @@
+package com.wynntils.mc.mixin;
+
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.wynntils.mc.EventFactory;
+import com.wynntils.mc.event.PlayerGhostArmorRenderEvent;
+import net.minecraft.client.renderer.MultiBufferSource;
+import net.minecraft.client.renderer.entity.layers.HumanoidArmorLayer;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.player.Player;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+@Mixin(HumanoidArmorLayer.class)
+public abstract class HumanoidArmorLayerMixin<T extends LivingEntity> {
+
+    @Inject(method = "render(Lcom/mojang/blaze3d/vertex/PoseStack;Lnet/minecraft/client/renderer/MultiBufferSource;ILnet/minecraft/world/entity/LivingEntity;FFFFFF)V", at = @At("INVOKE"), cancellable = true)
+    private void render(PoseStack matrixStack, MultiBufferSource buffer, int packedLight, T livingEntity, float limbSwing, float limbSwingAmount, float partialTicks, float ageInTicks, float netHeadYaw, float headPitch, CallbackInfo ci) {
+        if (!(livingEntity instanceof Player))
+            return;
+        PlayerGhostArmorRenderEvent event = EventFactory.onArmorRender(livingEntity);
```

Comment:
Cast livingEntity to Player here, where you made the check, and let the event and all subsequent users have a Player instead.